src/uast.cc

```diff
@@ -38,6 +42,52 @@ struct Nodes {
   int cap;
 };
 
+enum XPathType {
+  XPATHTYPE_UNDEFINED = 0,
+  XPATHTYPE_NODESET = 1,
+  XPATHTYPE_BOOLEAN = 2,
+  XPATHTYPE_NUMBER = 3,
+  XPATHTYPE_STRING = 4,
+  XPATHTYPE_POINT = 5,
+  XPATHTYPE_RANGE = 6,
+  XPATHTYPE_LOCATIONSET = 7,
+  XPATHTYPE_USERS = 8,
+  XPATHTYPE_XSLT_TREE = 9,
+};
+
+struct HashXPathType {
+  long operator()(const XPathType& xtype) const {
+    return static_cast<long>(xtype);
+  }
+};
+
+const std::unordered_map<XPathType, const char*, HashXPathType> Type2Str {
```

Comment:
An unordered_map may be a bit overkill here, but ok.

Reply:
Yeah, actually a vector of strings will server just the same since the indexing will be done by a number and there are no gaps in the enum.